elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -914,9 +914,18 @@ protected boolean checkRelation(String relationName) {
 
         /* If elide was configured for Elide 3.0 data store interface */
         if (requestScope.useFilterExpressions()) {
-            val = requestScope.getTransaction()
-                    .getRelation(obj, type, relationName, relationClass, dictionary,
-                            filterExpression, requestScope.getSorting(), requestScope.getPagination());
+            String path = requestScope.getPath();
+            String relationNameForSortAndPage = path.substring(path.lastIndexOf("/") + 1);
+            if (relationName.equals(relationNameForSortAndPage)) {
```

Comment:
I'm confused why we need these checks to extract the relationship from the path.  The state machine knows when pagination is needed and when it isn't.  I would assume the persistent resource would always call the non paginated version of getRelation.